src/test/java/org/folio/rest/impl/UsersAPIIT.java

```diff
@@ -91,20 +101,53 @@ static TenantAttributes tenantAttributes() {
   }
 
   static void postTenant() {
-    given().body(tenantAttributes()).
+    String id = given().body(tenantAttributes()).
     when().post("/_/tenant").
-    then().statusCode(201);
+    then().statusCode(201).
+    extract().
+    path("id");
+
+    waitForTenantInit(id);
+  }
+
+  //restassured has no way of doing asynchronous checking for tenant initialization
+  //without introducing more dependencies to support that.  Thus, a custom wait
+  //method to do this is neccessary.
+
+  private static void waitForTenantInit(String id) {
+
+    Boolean tenantSetupDone = false;
+    Integer maxTries = 10;
+    Integer tries = 0;
+    while (tenantSetupDone == false && tries <= maxTries) {
+      try {
+		    Thread.sleep(1000);
+	    } catch (InterruptedException e) {
+		    e.printStackTrace();
+      }
+      tenantSetupDone = given().when().get("/_/tenant/" + id).then().statusCode(200).extract().path("complete");
+      tries++;
+    }
+    
+    assertEquals(tenantSetupDone, true);
+
+    //if something went wrong internally with client setup,
+    //there will be an error attribute in the response body
+    given().when().get("/_/tenant/" + id).then().statusCode(200).body("$", not(hasKey("error")));
```

Comment:
In other modules (like mod-circulation) we have used libraries like Awaitility in order to avoid having to use loops and thread.sleep.

I think it could be worth doing that here as well.

Reply:
The only reason I did not do this is because it means introducing a new dependency into the module just to support this test-I was not sure that Would Be Looked Favorably Upon.  If that's fine, then I'm happy to use it-it certainly makes this task easier.

Reply:
I understand the caution over introducing new dependencies. FOLIO doesn't really have a clear stance on that.

In this case, I think it is preferable to using `thread.sleep`, the library is used elsewhere and it is only for testing.